src/Files.App.Storage/FtpStorage/FtpFileSystemService.cs

```diff
@@ -41,30 +42,37 @@ public async Task<bool> DirectoryExistsAsync(string path, CancellationToken canc
 			}
 		}
 
+		/// <inheritdoc/>
 		public async Task<ILocatableFolder> GetFolderFromPathAsync(string path, CancellationToken cancellationToken = default)
 		{
-			using var ftpClient = FtpHelpers.GetFtpClient(path);
-			await ftpClient.EnsureConnectedAsync(cancellationToken);
+			using var ftpClient = await GetClient(path);
+			var ftpPath = path.GetFtpPath();
 
-			var ftpPath = FtpHelpers.GetFtpPath(path);
 			var item = await ftpClient.GetObjectInfo(ftpPath, token: cancellationToken);
-			if (item is null || item.Type != FtpObjectType.Directory)
+			if (item is null || item.Type is not FtpObjectType.Directory)
 				throw new DirectoryNotFoundException("Directory was not found from path.");
 
 			return new FtpStorageFolder(ftpPath, item.Name);
 		}
 
+		/// <inheritdoc/>
 		public async Task<ILocatableFile> GetFileFromPathAsync(string path, CancellationToken cancellationToken = default)
 		{
-			using var ftpClient = FtpHelpers.GetFtpClient(path);
-			await ftpClient.EnsureConnectedAsync(cancellationToken);
+			using var client = await GetClient(path, cancellationToken);
+			var ftpPath = path.GetFtpPath();
 
-			var ftpPath = FtpHelpers.GetFtpPath(path);
-			var item = await ftpClient.GetObjectInfo(ftpPath, token: cancellationToken);
-			if (item is null || item.Type != FtpObjectType.File)
+			var item = await client.GetObjectInfo(ftpPath, token: cancellationToken);
+			if (item is null || item.Type is not FtpObjectType.File)
 				throw new FileNotFoundException("File was not found from path.");
 
 			return new FtpStorageFile(ftpPath, item.Name);
 		}
+
+		private static async Task<AsyncFtpClient> GetClient(string path, CancellationToken cancellationToken = default)
+		{
+			using AsyncFtpClient client = path.GetFtpClient();
```

Comment:
`using` statement here will dispose the client after the method exits. Remove it from here

Reply:
Besides, you could refactor the method in FtpStorable.GetFtpClientAsync() - to use a `string path` parameter, and remove this method